src/bin/clean.rs

```diff
@@ -0,0 +1,31 @@
+use std::io;
+use std::io::Write;
+use std::fs;
+use clap::{App, ArgMatches, SubCommand};
+use mdbook::MDBook;
+use mdbook::errors::Result;
+use get_book_dir;
+
+use std::error::Error;
+
+// Create clap subcommand arguments
+pub fn make_subcommand<'a, 'b>() -> App<'a, 'b> {
+    SubCommand::with_name("clean")
+        .about("Delete built book")
+}
+
+// Init command implementation
+pub fn execute(args: &ArgMatches) -> Result<()> {
+    let mut book_dir = get_book_dir(args);
+    book_dir.push("book");
+
+    match fs::remove_dir_all(book_dir.to_str().unwrap()) {
+        Ok(_) => (),
+        m @ _ => {
```

Comment:
What's the point of the `m @ _` here? It doesn't look like you're using the `m` anywhere...

Reply:
I thought about using it, passing somehow to Err.

Reply:
It'd be much better to do something like this:

```rust
fs::remove_dir_all(&book_dir).chain_err(|| "Unable to remove the build directory")?;
```

The `chain_err()` method lets you add extra context to an error ([docs]), which we then use to print something like this to the user:

```
Error: Unable to remove build directory
  Caused By: File not found
```

It also helps with your `return Err("message")` issue later on. You probably want to read the [error handling] chapter in *The Book* to get a better understanding of what's happening.

[docs]: https://docs.rs/error-chain/0.11.0/error_chain/#chaining-errors
[error handling]: https://doc.rust-lang.org/book/second-edition/ch09-02-recoverable-errors-with-result.html#propagating-errors